tests/data/trainercontroller/epoch-level-eval-loss-patience.yaml

```diff
@@ -0,0 +1,14 @@
+controller-metrics:
+  - name: eval_loss_window
+    class: HistoryBasedMetric
+    arguments:
+      window-size: 2
+controllers:
+  - name: epoch-level-eval-loss-patience
+    triggers:
+      - on_epoch_end
+    rule: len(eval_loss_window["metrics"]) > 0 and eval_loss_window["metrics"][-1]["eval_loss"] > 2
+    patience:
+      patience_threshold: 2
```

Comment:
Do we want to call it `patience_threshold` or `threshold` since patience is already part of the parent key

Reply:
let it be `patience_threshold`. Although redundant, it will be less confusing to user to relate to the concept...isnt it?

Reply:
Sure, But we should ensure we use either `-` or `_` consistently. We use `-` for `window-size` and `controller-metrics`, but we are using `_` here.

Reply:
Addressed. Except for `controller` name, `-` should not used anywhere.

Reply:
All `-` removed. Only using `_`.